metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1119,6 +1057,83 @@ void validateManualPartitionAssignment(List<Integer> assignment,
         }
     }
 
+    /**
+     * Iterate over a sequence of partitions and generate ISR changes and/or leader
+     * changes if necessary.
+     *
+     * @param context           A human-readable context string used in log4j logging.
+     * @param brokerToRemove    NO_LEADER if no broker is being removed; the ID of the
```

Comment:
What about this place? It seems that Instead of relying upon NO_LEADER, it might be clearer to make brokerToRemove and brokerToAdd optional int?

Reply:
I think it's easier to use NO_LEADER since then I don't have to special-case the comparisons (I can just compare with NO_LEADER and it will fail when examining ISR members, for example).